### ISSUE TITLE
qt-5 should not depend on llvmPackages_5 (?)

### ISSUE THREAD

**Issue by dnadlinger** [NixOS/nixpkgs#151874]
### Describe the bug

The Qt 5 derivations depend specifically on LLVM 5.0 when the stdenv cc is Clang:

https://github.com/NixOS/nixpkgs/blob/41513de1fd8947ef9ad6af465a20b564dc84b6f0/pkgs/development/libraries/qt-5/5.15/default.nix#L28

This is an ancient version, which is the first hard blocker for Qt on aarch64-darwin, where compiler-rt from that version doesn't build.

Is there a reason for requiring this old version?

### Steps To Reproduce
1. `nix --system aarch64-darwin nixpkgs#qt515.qtsvg
2. Note broken package warning for compiler-rt from llvmPackages_5.

### Expected behavior
I'm not sure why that `stdenvActual` sleight is necessary, but presumably, this should at least be `llvmPackages.stdenv`, such that `llvmPackages_11` is used on aarch64?

I've had a quick go at building a patched Qt 5.15 for aarch64, and while there are some fundamental toolchain issues to sort out still (even with the 5.15 aarch64 patch from Homebrew), sorting out the dependencies is at least a first step.

Marking Qt 5 itself as broken would also give much more intelligible error messages than compiler-rt-5 somewhere deep in the dependency tree of a project.

### Notify maintainers
@qknight @ttuegel  @periklis @bkchr @veprbl

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.2.0, macOS 12.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.5.1`
 - channels(dpn): `""`
 - channels(root): `"nixpkgs-22.05pre338661.3c52ea8c921"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by prusnak** [comment]
I tried both `llvmPackages.stdenv` and `llvmPackages_11.stdenv` and I am still getting the following error on aarch64-darwin:

```
error: unknown target CPU 'armv8.3-a+crypto+sha2+aes+crc+fp16+lse+simd+ras+rdm+rcpc'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, icelake-server, tigerlake, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, x86-64
```

(this of course happens with `llvmPackages_5.stdenv` as well.

**Comment by dnadlinger** [comment]
Yep, that's because for some reason qmake (?) also passes x86_64 as architecture on the clang command line, which of course makes the CPU specification invalid. I think the ancient llvmPackages dependency is a (small) issue regardless – unless there is a good reason to pull in the old version, of course.

**Comment by prusnak** [comment]
Here is my take on addressing this issue: https://github.com/NixOS/nixpkgs/pull/151903

**Comment by toonn** [comment]
As @SuperSamus points out in the linked PR #151903, this can be closed. It still transitively depends on multiple LLVMs for qt5.full, which is qt512.full AFAICT. Specifically 13.0.1, 15.0.7, 16.0.6 and 17.0.6. LLVM 16 is Darwin stdenv currently and I believe 17.0.6 is due to rustc.

Not sure it's worth digging into why there's so many but this particular issue was resolved by #151903.
